core/templates/dev/head/domain/utilities/url-interpolation.service.spec.ts

```diff
@@ -78,46 +75,46 @@ describe('URL Interpolation Service', function() {
     );
   });
 
-  it('should throw an error for erroneous URLs', function() {
-    expect(uis.interpolateUrl).toThrow(
-      new Error('Invalid or empty URL template passed in: \'undefined\''));
-    expect(uis.interpolateUrl.bind(null, null, {})).toThrow(
+  it('should throw an error for erroneous URLs', () => {
+    expect(uis.interpolateUrl.bind(uis, null, null)).toThrow(
```

Comment:
I'm able to pass the tests for this service now :), though I'm still not clear why we need the bind call here? 

Bind function is used to bind the `this` keyword in the function(here interpolateUrl) with the first argument(here uis). Since, we are calling uis.interpolateUrl() `this` in interpolateUrl would be `uis` itself(I've checked this and it's fine) but `toThrow` is not able to catch the expected error, I don't know why?

What is bind doing here such that it's able to catch the error now whereas couldn't do without bind?

cc/ @seanlip @kevinlee12 @ankita240796 @Showtim3

Reply:
Perhaps relevant? https://stackoverflow.com/questions/42976291/expect-not-tothrow-function-with-arguments-jasmine

Also, just to check, what were you doing before? Did you have expect(uis.interpolateUrl(null, null))? Because that won't work, since the contents of "expect" would be the result of the function evaluation in this case, rather than a function that is waiting to be executed.

Reply:
Yeah, that link is relevant :).

Yes, I was doing requires function to be passed as argument  -- got it now, thanks @seanlip!